lib/tracecontext/tracestate.js

```diff
@@ -40,17 +47,13 @@ class TraceState {
     for (const [key, value] of Object.entries(defaultValues)) {
       this.setValue(key, value)
     }
-
-    // if no logger, then fallback to agent singleton
-    logger = logger || require('../..').logger
-    this.logger = logger
```

Comment:
In the *extreme* case, it is possible for the `agent.logger` to be replaced with a new logger object from a central config update *while* a TraceState instance is holding a ref to it.

Reply:
@trentm No objection to this change -- but a question to make sure I understand what we're fixing here.  

Is the problem with the previous code that 

1. A TraceState object is instantiated
2. We receive a message from APM Server that there's a new log level, so a new logger is instantiated
3. However, the TraceState object will still hold a reference to the old logger, meaning it will continue to log at the "wrong"/"old" level for the remainder of the trace's life?

i.e. is the problem (in the extreme case) a trace state header will log at the wrong level?  Or are there other problems with the header holding on to the old object reference?

Reply:
Yes, that (your 1. 2. 3.) is the (theoretical) problem.  With the current logger lib and only logging to stdout/stderr, yes the only issue is potentially logging at the wrong level.  Going *more* extreme, if we were to support a new logging framework (e.g. the potential move to `pino`) **and** we were to support logging to a file or some stream, the TraceState instance holding on to that logger could delay resource cleanup. I'm stretching now. :)